registry/remote/repository.go

```diff
@@ -992,16 +1046,33 @@ func verifyContentDigest(resp *http.Response, expected digest.Digest) error {
 	if digestStr == "" {
 		return nil
 	}
-	contentDigest, err := digest.Parse(digestStr)
-	if err != nil {
+	if contentDigest, err := digest.Parse(digestStr); err != nil {
 		return fmt.Errorf("%s %q: invalid response Docker-Content-Digest: %s", resp.Request.Method, resp.Request.URL, digestStr)
-	}
-	if contentDigest != expected {
+	} else if contentDigest != expected {
 		return fmt.Errorf("%s: mismatch digest: %s", expected, contentDigest)
 	}
 	return nil
 }
 
+// verifyContentDigests is a superset of verifyContentDigest; it takes in the actual (calculated) digest, and
+// verifies against that, the other two digests, both of which may or may not be set to the empty digest.  If empty,
+// there is no validation error, however is not empty, it ensures that the given digest is valid (matches the actual
+// digest)
+func verifyContentDigests(actualDigest digest.Digest, digests map[string]digest.Digest) error {
+	for src, d := range digests {
+		if len(d.String()) == 0 {
+			continue
+		}
+		if string(actualDigest) != string(d) {
```

Comment:
`if actualDigest != d` would be enough.

Reply:
Same as above, this was only for communication to the reader, to remind the reader that this `string` is actually a `digest.Digest`.  I'll change though, since preference is still for the `go-digest` change to go through, and then we can use `if d.IsIdenticalTo(actualDigest)`.